### PR TITLE
fix: miscellaneous issues in table widgets

### LINKED ISSUE #17307: [Bug]: Save/Discard option is present in the filters option which is causing the table widget to crash.
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Save/Discard option is present in the filters option which is causing the table widget to crash.

https://user-images.githubusercontent.com/109572422/195045841-07da414b-251d-4d3e-b703-7b847ec4a7c0.mov

### Steps To Reproduce

1. D&D a Table widget
2. Enable column(s) to be editable
3. In the filters option, notice that the save/discard option is present. And on clicking on it, the table crashes.

### Public Sample App

_No response_

### Version

Cloud

### LINKED ISSUE #17502: [Bug]: Cursor jumps to start while editing cell in Table Widget
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When we are typing in cell edit mode inside a table widget with cell wrapping on, the cursor sometimes jumps to the start of the input field.

### Steps To Reproduce

1. Drag and drop a table widget
2. Make a Plain Text cell editable
3. Turn on Cell Wrapping for that column and another Plain Text column with text spanning at least 3 lines
4. Click on edit icon in the cell and keep typing
5. The cursor jumps to the initial position of the input
6. This happens only once per cell. Editing a second time does not show this issue

To reproduce the issue, edit the `id` column in the sample app provided.

### Public Sample App

https://app.appsmith.com/app/untitled-application-1/page1-6343a43a89020c7ac296953e

### Version

Appsmith v1.8.4

### LINKED ISSUE #17797: [Bug]: Retain filter in table widget when table data changes but schema remains same
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When the table data of the table widget changes we reset the applied filters regardless of whether the schema of the tabledata changes or not. We should only reset the filters if the schema changes.

### Steps To Reproduce

1. Drag and drop a table
2. apply a filter
3. Use inline editing to update a cell value.
4. refresh the query bound to the table widget.
5. notice that applied filters got reset since the table data changed with the update.

### Public Sample App

_No response_

### Version

cloud

Front logo [Front conversations](https://app.frontapp.com/open/top_5emx8)

### PR DESCRIPTION
**Pull request by sbalaji1192**
## Description

- fix: remove the edit action column from the filter dropdown
- fix: reset table filters only when table data schema changes

Fixes #17797 
Fixes #17502 
Fixed #17307

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
- Manual
- Cypress

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

**Comment by chandannkumar** [comment]
Intermittent behavior:
- Cursor is not visible on view mode when Table widget is binded with Queries, API.
This is observed in Chrome browser only (Windows machine)
https://www.loom.com/share/164bdf4b0c1c4c7389331f7df4e96823

**Comment by chandannkumar** [comment]
Since the above callouts is intermittent and cannot be reproducible every time.
We will get back to it if actual users also face the same issue on Chrome browser.

Moving this PR to done, as issue connected to this PR are fixed and tested

File: app/client/src/widgets/TableWidgetV2/component/cellComponents/InlineCellEditor.tsx [hunk 1]
```diff
@@ -191,17 +192,15 @@ export function InlineCellEditor({
...
-        if (cursorPos < value.length) {
-          inputRef.current.selectionEnd = cursorPos;
-        }
+      if (cursorPos < value.length) {
+        inputRef.current.selectionEnd = cursorPos;
       }
-    }, 0);
-  }, [cursorPos, inputRef.current, value]);
+    }
+  }, [multiline]);
```
Comment (ashit-rath) [inline app/client/src/widgets/TableWidgetV2/component/cellComponents/InlineCellEditor.tsx:203]: Why are we changing the used dependencies with `multiline` which is not present in the `useLayoutEffect` implementation?

File: app/client/src/widgets/TableWidgetV2/component/cellComponents/PlainTextCell.tsx [hunk 2]
```diff
@@ -158,15 +173,17 @@ function PlainTextCell(props: RenderDefaultPropsType & editPropertyType) {
...
-      contentRef.current?.offsetHeight / CELL_WRAPPER_LINE_HEIGHT > 1;
+  const [isMultiline, setIsMultiline] = useState(false);
+
+  useEffect(() => {
+    if (isCellEditMode) {
+      fastdom.measure(() => {
+        setIsMultiline(getContentHeight(contentRef));
+      });
+    }
+  }, [value, isCellEditMode]);
```
Comment (ashit-rath) [inline app/client/src/widgets/TableWidgetV2/component/cellComponents/PlainTextCell.tsx:184]: Why add `value` here?

File: app/client/src/widgets/TableWidgetV2/component/cellComponents/InlineCellEditor.tsx (hunk 1 above)
Comment (sbalaji1192) [reply app/client/src/widgets/TableWidgetV2/component/cellComponents/InlineCellEditor.tsx:203]: We want to place the cursor when the input changes to textArea and vice versa which the multiline boolean value dictates.

File: app/client/src/widgets/TableWidgetV2/component/cellComponents/PlainTextCell.tsx (hunk 2 above)
Comment (sbalaji1192) [reply app/client/src/widgets/TableWidgetV2/component/cellComponents/PlainTextCell.tsx:184]: As you can see we're checking the content height to determine whether we need to use an input (for single line) or textarea(multiline). Content height only changes when the value changes.

**Review by ashit-rath — APPROVED** [review]

**Review by ashit-rath — APPROVED** [review]

**Pull request merged**